File: Day12/imagekit.py
from importlib.metadata import files
from fastapi import HTTPException
import os
from imagekitio import ImageKit

imagekit = ImageKit(
    private_key=os.getenv("IMAGEKIT_PRIVATE_KEY")
)

URL_ENDPOINT = os.getenv("IMAGEKIT_ENDPOINT_URL")

async def upload_file(file, file_id):
    try:
        file_bytes = await file.read()
        response = imagekit.files.upload(
            file = file_bytes,
            file_name = file_id,
            folder="/files"
        )

        return {
            "file_id": response.fileId,
            "file_url": response.url
            }
    
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))


def delete_file(imagekit_id):
    try:
        response = imagekit.files.delete(
            file_id = imagekit_id
        )
        return response
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"File not deleted {e}")

File: Day14/src/services/imagekit.js
import ImageKit from "@imagekit/nodejs";
import dotenv from "dotenv";

dotenv.config();

const imagekit = new ImageKit({
    privateKey: process.env.IMAGEKIT_PRIVATE_KEY,
    publicKey: process.env.IMAGEKIT_PUBLIC_KEY,
    urlEndpoint: process.env.IMAGEKIT_URL_ENDPOINT,
});

const upload_file = async (filename, file) => {
    try {
        const result = await imagekit.files.upload({
            file: file.buffer.toString("base64"),
            fileName: filename,
            folder: "/files",
        });

        return { "fileUrl": result.url, "imagekitId": result.fileId };
    } catch (error) {
        console.log(error);
        return error;
    }
};

const delete_file = async (imagekitId) => {
    try {
        const result = await imagekit.files.delete(imagekitId);
        return result;
    } catch (error) {
        console.log(error);
        return error;
    }
};

export { upload_file, delete_file };